File: benchmarks/stress.go
// Go Extreme Stress Test
package main

import (
	"fmt"
	"os"
	"runtime"
	"strconv"
)

const ALLOC_SIZE = 1048576 // 1MB

func testAllocations(count int) int {
	runtime.GC()

	slices := make([][]byte, 0, count)
	success := 0

	defer func() {
		if r := recover(); r != nil {
			fmt.Printf("  PANIC at allocation #%d: %v\n", success, r)
		}
	}()

	for i := 0; i < count; i++ {
		data := make([]byte, ALLOC_SIZE)
		for j := 0; j < ALLOC_SIZE; j += 4096 {
			data[j] = byte(i)
		}
		slices = append(slices, data)
		success++
	}

	return success
}

func main() {
	target := 1000
	if len(os.Args) > 1 {
		if t, err := strconv.Atoi(os.Args[1]); err == nil {
			target = t
		}
	}

	fmt.Printf("Go Stress Test: %d x 1MB allocations\n", target)

	success := testAllocations(target)

	fmt.Printf("Result: %d/%d allocations succeeded\n", success, target)
	fmt.Printf("Memory used: %d MB\n", success)

	if success < target {
		fmt.Printf("STATUS: FAILED at %d\n", success)
		os.Exit(1)
	}
	fmt.Println("STATUS: PASSED")
}

File: benchmarks/stress.rs
// Rust Extreme Stress Test
use std::env;

const ALLOC_SIZE: usize = 1048576; // 1MB

fn test_allocations(count: usize) -> usize {
    let mut vecs: Vec<Vec<u8>> = Vec::with_capacity(count);
    let mut success = 0;
    
    for i in 0..count {
        match std::panic::catch_unwind(|| {
            vec![0u8; ALLOC_SIZE]
        }) {
            Ok(v) => {
                vecs.push(v);
                success += 1;
            }
            Err(_) => {
                println!("  FAILED at allocation #{}", i);
                break;
            }
        }
    }
    
    success
}

fn main() {
    let args: Vec<String> = env::args().collect();
    let target: usize = if args.len() > 1 {
        args[1].parse().unwrap_or(1000)
    } else {
        1000
    };
    
    println!("Rust Stress Test: {} x 1MB allocations", target);
    
    let success = test_allocations(target);
    
    println!("Result: {}/{} allocations succeeded", success, target);
    println!("Memory used: {} MB", success);
    
    if success < target {
        println!("STATUS: FAILED at {}", success);
        std::process::exit(1);
    }
    println!("STATUS: PASSED");
}
